src/libstore/derivations.cc

```diff
@@ -154,31 +155,76 @@ StorePath writeDerivation(Store & store,
 }
 
 
+namespace {
+/* This mimics std::istream to some extent. We use this much smaller implementation
+   instead of plain istreams because the sentry object overhead is too high. */
+struct ParseContext {
+    std::string_view str;
```

Comment:
`str` is rather ambiguous. How about either

```suggestion
    std::string_view remaining;
```

or

```suggestion
    std::string_view view;
```